plugins/cue_points/quiz/services/QuizUserEntryService.php

```diff
@@ -51,12 +51,38 @@ public function submitQuizAction($id)
 		}
 
 		$c = new Criteria();
+
 		$c->add(CuePointPeer::ENTRY_ID, $dbUserEntry->getEntryId(), Criteria::EQUAL);
 		$c->add(CuePointPeer::TYPE, QuizPlugin::getCoreValue('CuePointType', QuizCuePointType::QUIZ_QUESTION));
 		$dbUserEntry->setNumOfQuestions(CuePointPeer::doCount($c));
 		$dbUserEntry->setStatus(QuizPlugin::getCoreValue('UserEntryStatus', QuizUserEntryStatus::QUIZ_SUBMITTED));
 		$dbUserEntry->save();
 
+		if ($kQuiz->getShowCorrectAfterSubmission())
+		{
```

Comment:
All this block should go under a function that returns an array of correct optional answers.